addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -2246,10 +2253,14 @@ var SnippetsMenu = Widget.extend({
                 // When noCheck is true, we only check the exclude.
                 return true;
             }
-            // `o_editable_media` bypasses the `o_not_editable` class.
-            if (this.classList.contains('o_editable_media')) {
+            // `o_editable_media` bypasses the `o_not_editable` class except for
+            // drag & drop.
+            if (!forDrop && this.classList.contains('o_editable_media')) {
                 return shouldEditableMediaBeEditable(this);
             }
+            if (isChildren) {
+                return !$(this).is('.o_not_editable *');
+            }
             return !$(this).is('.o_not_editable :not([contenteditable="true"])');
```

Comment:
If the `isChildren` part above is actually needed, I wonder if this other part is enough too. What if you change the drop-near which allows to add the text highlight snippet in the DOM (see `so_content_addition_selector`) by a `drop-in=".row > div"`? Does it work (= you can still add inline snippets in columns but not in the o_not_editable one)?

Reply:
mhh... indeed, with the following change, the bug is still present
```diff
<div id="so_content_addition"
         t-att-data-selector="so_content_addition_selector"
-        t-attf-data-drop-near="p, h1, h2, h3, ul, ol, .row > div > img, #{so_content_addition_selector}"
-        data-drop-in=".content, nav"/>
+        data-drop-in=".row > div"/>
```
when the `this` is the `.o_not_editable` itself, it doesn't match `.o_not_editable :not([contenteditable="true"])` as the element is the `.o_not_editable`. So yes indeed for the drop-in we have to check that the element doesn't match `.o_not_editable :not([contenteditable="true"])` **and** is not itself a `.o_not_editable, [contenteditable="false"]` I guess @qsm-odoo ?

Reply:
No need of the `[contenteditable="false"]` part I think. Mais effectivement "valid element if not o_not_editable and not the child of a o_not_editable, unless I am contenteditable="true".

Reply:
Ok I made [this change](https://github.com/odoo/odoo/compare/47935f4aec826e4343df52d166157b15c0393292..c83ebb42a3c62b10cc8bba89fae68c15a46068ba) then @qsm-odoo :heavy_check_mark: